### PR TITLE
Query: Refactors the EnableOptimisticDirectExecution flag to be public

### PR DESCRIPTION
**Pull request by akotalwar**
# Pull Request Template

## Description

Made the EnableOptimisticDirectExecution flag a public flag so that customers can now enable this pipeline.

## Type of change

Please delete options that are not relevant.

- [] New feature (non-breaking change which adds functionality)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public class QueryRequestOptions : RequestOptions
         /// <value>
         /// Direct (optimistic) execution offers improved performance for several kinds of queries such as a single partition streaming query.
         /// </value>
-        internal bool EnableOptimisticDirectExecution { get; set; }
+        public bool EnableOptimisticDirectExecution { get; set; }
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:52]: Are there any aka.ms or doc links we can point customers to to learn on which scenarios they should use this?

Should we update https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/performance-tips-query-sdk?tabs=v3&pivots=programming-language-csharp

Comment (akotalwar) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:52]: We dont have anything yet, but James and I are working on adding documentation for it. Should have it out soon.

**Review by sboshra — APPROVED** [review]
:shipit:

**Review by sboshra — APPROVED** [review]
:shipit:

**Pull request merged**